### DISCUSSION TITLE
App exits after failed to obtain initial connection

### DISCUSSION THREAD

**Discussion by pli2019** [playframework/playframework#12665] [Q&A]
Hi, 

I have an app written in playframework 2.3.9 and it is using bonecp as the jdbc pool.  During the startup, if the database is not available, the app will exit without any retry.  I checked the bonecp source codes and it specified no retry at initial connection. I want to ask if there is any way/configuration to retry the database initial connection from the playframe side?  Thanks

```
api  | Play server process ID is 1
api  | [error] c.j.b.h.AbstractConnectionHook - Failed to obtain initial connection Sleeping for 0ms and trying again. Attempts left: 0. Exception: java.net.UnknownHostException: postgres.Message:The connection attempt failed.
api  | Oops, cannot start the server.
api  | Configuration error: Configuration error[Cannot connect to database [testdb]]
api  |  at play.api.Configuration$.play$api$Configuration$$configError(Configuration.scala:94)
api  |  at play.api.Configuration.reportError(Configuration.scala:743)
api  |  at play.api.db.BoneCPPlugin$$anonfun$onStart$1.apply(DB.scala:247)
api  |  at play.api.db.BoneCPPlugin$$anonfun$onStart$1.apply(DB.scala:238)
api  |  at scala.collection.immutable.List.map(List.scala:273)
api  |  at play.api.db.BoneCPPlugin.onStart(DB.scala:238)
api  |  at play.api.Play$$anonfun$start$1$$anonfun$apply$mcV$sp$1.apply(Play.scala:91)
```

**Comment by mkurz** [comment]
Play 2.3.9 is not supported anymore. I don't think we can help you...
I mean, I was looking into that for 1 minute and found that _maybe_ you can tweak the settings
```
db.default.acquireRetryAttempts
db.default.acquireRetryDelay
```
which are set here: https://github.com/playframework/playframework/blob/2.3.9/framework/src/play-jdbc/src/main/scala/play/api/db/DB.scala#L383-L384

But I somehow doubt this will help.

Good luck. And time to upgrade to the latest Play releases.
